runtime/doc/various.txt

```diff
@@ -250,12 +250,17 @@ g8			Print the hex values of the bytes used in the
 			A '|' in {cmd} is passed to the shell, you cannot use
 			it to append a Vim command.  See |:bar|.
 
-			If {cmd} contains "%" it is expanded to the current
-			file name.  Special characters are not escaped, use
-			quotes to avoid their special meaning: >
+			If {cmd} contains "%" or "#" it is expanded to the
+			appropriate file name (|:_%| |:_#|).  Use escaping to
+			avoid this expansion: >
+				:!ls \%
+<			This should always work, but it's more typing: >
+				:exe "!ls " . escape("the#ofthecounting.txt", "%#")
```

Comment:
Are these two lines needed?

Reply:
The string "the#ofthecounting.txt" confuses me.
`:!ls the\#ofthecounting.txt` is enough for this case.

I think that using `escape()` here is useful only when constructing a command line by using a variable which may contain `%` or `#`.
E.g.:

```vim
let filename = input('input a file name: ')
exe "!ls " . escape(filename, "%#")
```

Reply:
This is the use case that led me to this text. My use case is more like:

```
exe "grep ". escape(@/, "%#")
```

but that seems unclear to me what would go wrong if you didn't use escape(). Do you think that's a better example?

Reply:
Ah, this is a good example.